torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -57,7 +57,7 @@ void checkImplicitTensorToNum(at::Tensor t, bool toInt) {
         "Cannot input a tensor of dimension other than 0 as a scalar argument");
   }
   if (toInt &&
-      !isIntegralType(autograd::as_variable_ref(t).data().scalar_type())) {
+      !isIntegralType(t.scalar_type())) {
```

Comment:
It's okay to remove the `.data()` call here, because we only care about getting `scalar_type()` of `t`'s data tensor, and that operation doesn't accumulate gradient.